airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -486,6 +487,11 @@ def run_load(self,
                 'writeDisposition': write_disposition,
             }
         }
+        
+        # if it is a partitioned table ($ is in the table name) add partition load option
+        if '$' in destination_project_dataset_table:
+            configuration['load']['timePartitioning'] = dict(type='DAY')
```

Comment:
Maybe you would want to add support for the other two nested options of `timePartitioning`: `expirationMs` and `field`?

Reply:
I agree with you on this, will try to add it sometime this week